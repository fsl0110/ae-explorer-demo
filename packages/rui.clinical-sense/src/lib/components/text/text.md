# Text

### Example

```jsx
<Text color="blue">I'm a Text</Text>
```
