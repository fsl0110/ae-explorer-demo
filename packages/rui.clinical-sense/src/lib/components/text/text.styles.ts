import styled from "styled-components";

export interface Props {
  color?: string;
}
export const TextStyles = styled.span<Props>`
  color: ${(props) => props.color || "#fff"};
`;
