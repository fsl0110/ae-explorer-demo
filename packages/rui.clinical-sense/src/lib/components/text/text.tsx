import React, {FC, ReactNode} from "react";
import {TextStyles} from "./text.styles";

export interface Props {
  /**
   * Color of the text.
   * @default #fff
   */
  color?: string;

  /**
   * Children of the text.
   */
  children: ReactNode | ReactNode[];
}
export const Text: FC<Props> = ({...rest}) => <TextStyles {...rest} />;
