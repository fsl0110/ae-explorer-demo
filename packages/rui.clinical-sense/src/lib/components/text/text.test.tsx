import React from "react";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import {Text, Props} from "./text";

const testCases: Array<[string, Props]> = [
  [
    "Text default",
    {
      children: "some text",
    },
  ],
  [
    "Text color blue h1",
    {
      color: "blue",
      children: "some text",
    },
  ],
];

describe("<Text>", () => {
  testCases.forEach(([item, options]) => {
    it(item, () => {
      expect(toJson(shallow(<Text {...options} />))).toMatchSnapshot();
    });
  });
});
