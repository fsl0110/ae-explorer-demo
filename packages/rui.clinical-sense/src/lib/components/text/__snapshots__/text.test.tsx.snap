// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text> Text color blue h1 1`] = `
<styled.span
  color="blue"
>
  some text
</styled.span>
`;

exports[`<Text> Text default 1`] = `
<styled.span>
  some text
</styled.span>
`;
